### PR TITLE
Optimize user submission flow

### PR DESCRIPTION
**Pull request by bwalsh**
This PR has several enhancements:
* `auth`:  the user will only need to maintain a single credential ~/.gen3/gen3_client_config.ini per gen3-client instructions
  * a '--profile' option now exists on all commands

* `metadata generation`:  the system will default to filtering uploaded files to those files in the current manifest (that have been successfully uploaded), previously the system considered all files uploaded to indexd `--source [manifest|indexd]`

* `metadata submission`:  the system will now automatically generate and submit metadata. Additionally the system will default to immediately returning the job identifier.  The user can then query job status.

  * The user can override this behavior on the `gen3_util files manifest upload` command:
    * a) not to generate and submit `--meta_data`
    * b) choose to wait for the job to complete `--wait`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -15,20 +15,25 @@ PyJWT
...
+PyYAML>=6.0.1
+
 
 # function caching
 # joblib
 # IO
 
 # FHIR Model
-fhir.resources
+fhir.resources==7.0.2
```
Comment (matthewpeterkort) [inline requirements.txt:32]: fhir.resources requirement on line 12 needs to be deleted because duplicate entry. Same for pyJwt library on line 14.

**Comment by matthewpeterkort** [comment]
Had to go digging to figure out that the gen3_util by default looks for the first profile in the ~/.gen3/gen3_client_config.ini if you don't specify a profile using the flag. This should be somewhere in the docs or the very least in the README of this repo for developers wondering why their the authentication pytests are failing. 

1 test still fails:
```pytest tests/integration/test_project.py::test_project_bad_ping```

```gen3_util % gen3_util ping --cred SDGDG
/Users/peterkor/Desktop/NEWGEN3/compose-services-training/gen3_util_test/gen3_util/submission/lib/python3.9/site-packages/urllib3/__init__.py:34: NotOpenSSLWarning: urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020
  warnings.warn(
Usage: gen3_util ping [OPTIONS]
Try 'gen3_util ping --help' for help.

Error: no such option: --cred
```
Also good to note that ```gen3_util ping --help``` doesn't return any meaningful help information to help the user

File: gen3_util/access/__init__.py [hunk 2]
```diff
@@ -9,21 +9,35 @@ def _ensure_auth(auth, config):
...
     if mine:
         # returns a list of dicts
         # https://github.com/uc-cdis/requestor/blob/master/src/requestor/routes/query.py#L200
-        return auth.curl('/requestor/request/user').json()
+        url = '/requestor/request/user'
+        parms = []
+        if active:
+            parms.append("active")
+        if len(parms) > 0:
+            url = url + "?" + "&".join(parms)
```
Comment (matthewpeterkort) [inline gen3_util/access/__init__.py:40]: Guessing these "parms" additions are for harnessing the full functionality of the requestor API endpoint?

File: gen3_util/config/__init__.py [hunk 3]
```diff
@@ -9,12 +9,37 @@
 from pydantic import BaseModel
 
 
-def ensure_auth(refresh_file: [pathlib.Path, str] = None, validate: bool = False) -> Gen3Auth:
+def gen_client_ini_path() -> pathlib.Path:
+    """Return path to gen3-client ini file."""
+    return pathlib.Path(pathlib.Path.home() / ".gen3" / "gen3_client_config.ini")
```
Comment (matthewpeterkort) [inline gen3_util/config/__init__.py:22]: This functionality should be a bit better documented, especially for people who have expired gen3_client configs in their gen3_client_config.ini file

File: gen3_util/files/remover.py [hunk 4]
```diff
@@ -1,7 +1,22 @@
...
-from gen3_util.config import Config
+from json import JSONDecodeError
 
+from gen3.file import Gen3File
 
-def rm(config: Config):
+from gen3_util.config import Config, ensure_auth
+
+
+def rm(config: Config, object_id: str) -> dict:
```
Comment (matthewpeterkort) [inline gen3_util/files/remover.py:8]: Is there a test for this function?

File: gen3_util/meta/skeleton.py [hunk 5]
```diff
@@ -54,55 +55,79 @@ def update_document_reference(document_reference, index_record):
...
+    """Read files uploaded to indexd or the local manifest and create a skeleton graph for document and ancestors from a set of identifiers.
 
     Args:
         config:
         project_id:
         output_path:
         overwrite: check for existing records and skip if found
+        source: indexd or manifest
     """
+    from gen3_util.files.manifest import ls as manifest_ls  # TODO fix circular import
```
Comment (matthewpeterkort) [inline gen3_util/meta/skeleton.py]: How to fix this?

**Review by matthewpeterkort — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (bwalsh) [reply requirements.txt:32]: done

File: gen3_util/access/__init__.py (hunk 2 above)
Comment (bwalsh) [reply gen3_util/access/__init__.py:40]: done

File: gen3_util/config/__init__.py (hunk 3 above)
Comment (bwalsh) [reply gen3_util/config/__init__.py:22]: done

File: gen3_util/files/remover.py (hunk 4 above)
Comment (bwalsh) [reply gen3_util/files/remover.py:8]: added

File: gen3_util/meta/skeleton.py (hunk 5 above)
Comment (bwalsh) [reply gen3_util/meta/skeleton.py]: fixed

**Pull request merged**